pkg/client/unversioned/discovery_client.go

```diff
@@ -71,27 +69,18 @@ func apiVersionsToAPIGroup(apiVersions *unversioned.APIVersions) (apiGroup unver
 	return
 }
 
-func (d *DiscoveryClient) get(url string) (resp *http.Response, err error) {
-	req, err := http.NewRequest("GET", url, nil)
-	if err != nil {
-		return nil, err
-	}
-	return d.httpClient.Do(req)
-}
-
 // ServerGroups returns the supported groups, with information like supported versions and the
 // preferred version.
 func (d *DiscoveryClient) ServerGroups() (apiGroupList *unversioned.APIGroupList, err error) {
 	// Get the groupVersions exposed at /api
-	url := d.baseURL
+	url := url.URL{}
 	url.Path = "/api"
```

Comment:
You can probably use a pure JSON Codec instead of using our built in one
for handling objects.  Unversioned JSON codec is better than being a nil
codec

On Tue, Oct 20, 2015 at 6:48 PM, Daniel Smith <notifications@github.com>
wrote:

> In pkg/client/unversioned/discovery_client.go
> <https://github.com/kubernetes/kubernetes/pull/15973#discussion_r42565014>
> :
>
> >  // ServerGroups returns the supported groups, with information like supported versions and the
> >  // preferred version.
> >  func (d *DiscoveryClient) ServerGroups() (apiGroupList *unversioned.APIGroupList, err error) {
> >  	// Get the groupVersions exposed at /api
> > -	url := d.baseURL
> > +	url := url.URL{}
> >  	url.Path = "/api"
>
> just use "/api" directly, no need to make the url.URL if you're not
> actually using it for anything complicated.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/15973/files#r42565014>.
>